crates/ide_assists/src/handlers/convert_comment_block.rs

```diff
@@ -0,0 +1,416 @@
+
+    let output = format!("{}\n{}\n{}*/", block_prefix, block_comment_body, indentation.to_string());
+
+    acc.add(
+        AssistId("line_to_block", AssistKind::RefactorRewrite),
+        "Replace line comments with a single block comment",
+        target,
+        |edit| edit.replace(target, output),
+    )
+}
+
+/// The line -> block assist can  be invoked from anywhere within a sequence of line comments.
+/// relevant_line_comments crawls backwards and forwards finding the complete sequence of comments that will
+/// be joined.
+fn relevant_line_comments(comment: &ast::Comment) -> Vec<Comment> {
+    // The prefix identifies the kind of comment we're dealing with
+    let prefix = comment.prefix();
+    let same_prefix = |c: &ast::Comment| c.prefix() == prefix;
+
+    // These tokens are allowed to exist between comments
+    let skippable = |not: &SyntaxElement| {
+        not.clone()
+            .into_token()
+            .and_then(Whitespace::cast)
+            .map(|w| !w.spans_multiple_lines())
+            .unwrap_or(false)
+    };
+
+    // Find all preceding comments (in reverse order) that have the same prefix
+    let prev_comments = comment
+        .syntax()
+        .siblings_with_tokens(Direction::Prev)
+        .filter(|s| !skippable(s))
+        .map(|not| not.into_token().and_then(Comment::cast).filter(same_prefix))
+        .take_while(|opt_com| opt_com.is_some())
+        .filter_map(identity)
+        .skip(1); // skip the first element so we don't duplicate it in next_comments
+
+    let next_comments = comment
+        .syntax()
+        .siblings_with_tokens(Direction::Next)
+        .filter(|s| !skippable(s))
+        .map(|not| not.into_token().and_then(Comment::cast).filter(same_prefix))
+        .take_while(|opt_com| opt_com.is_some())
+        .filter_map(identity);
+
+    let mut comments: Vec<_> = prev_comments.collect();
+    comments.reverse();
+    comments.extend(next_comments);
+    comments
+}
+
+// Line comments usually begin with a single space character following the prefix as seen here:
+//^
+// But comments can also include indented text:
+//    > Hello there
+//
+// We handle this by stripping *AT MOST* one space character from the start of the line
+// This has its own problems because it can cause alignment issues:
+//
+//              /*
+// a      ----> a
+//b       ----> b
+//              */
+//
+// But since such comments aren't idiomatic we're okay with this.
+fn line_comment_text(indentation: IndentLevel, comm: ast::Comment) -> String {
+    let contents = trim_one(comm.text().strip_prefix(comm.prefix()).unwrap()).to_owned();
+
+    // Don't add the indentation if the line is empty
+    if contents.is_empty() {
+        contents
+    } else {
+        indentation.to_string() + &contents
+    }
+}
+
+fn trim_one(text: &str) -> &str {
+    if text.starts_with(' ') {
+        &text[1..]
+    } else {
+        text
+    }
+}
```

Comment:
I think this is https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix?

Reply:
This is slightly different because it "succeeds" whether or not the pattern (`' '`) exists. strip_prefix will return null if it's not there. So instead we'd need something like: `text.strip_prefix(' ').unwrap_or(text)`

Reply:
Using `text.strip_prefix(' ').unwrap_or(text)` seems fine to me